### ISSUE TITLE
Provide interoperation with the Contrast Security Java agent

### ISSUE THREAD

**Issue by argonotix** [open-telemetry/opentelemetry-java-instrumentation#7099]
Recently it's been brought to our attention that the interoperability between OpenTelemetry's Java agent and the  [Contrast Security](https://www.contrastsecurity.com/) Java Agent has shown some compatibility issues.  Notably, if OpenTelmetry is placed first in the command line arguments, there is a significant slowdown in the target application.  If Contrast is placed first, the retransformation of classes by OpenTelemetry causes the target application to crash due to a misplacement of the proper `ClassLoader`.

For Contrast's side, we've already excluded all transformations of OpenTelemetry classes, and I'm suggesting we add a similar exclusion to [IgnoredTypesMatcher](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ignore/IgnoredTypesMatcher.java) with something simple like:

```java
if (name.startsWith("com.contrastsecurity") {
 return true;
}
```

Excluding Contrast classes relieves both the crash and the performance issues in my testing.

**Comment by trask** [comment]
hi @argonotix!

check out #1534

tldr:

> this is not something we plan to officially test/support, but we will accept PRs to make it run better alongside other agents

this would be the better location for the new ignore:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/2152feaabf301c495950ebb58f2f338e86d83c9d/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ignore/GlobalIgnoredTypesConfigurer.java#L31-L42

**Comment by argonotix** [comment]
@trask Perfect, thanks very much. Will do just that.
